src/main/typescript/wcardinal/ui/d-slider-horizontal.ts

```diff
@@ -25,38 +24,17 @@ export class DSliderHorizontal<
 	OPTIONS extends DSliderHorizontalOptions<THEME> = DSliderHorizontalOptions<THEME>
 > extends DSlider<THEME, OPTIONS> {
 
-	protected newTrack(): DSliderTrack {
-		return new DSliderTrackHorizontal();
+	protected newTrack( options?: OPTIONS ): DSliderTrack {
+		return new DSliderTrackHorizontal( options && options.track );
 	}
 
-	protected newTrackSelected(): DSliderTrack {
-		return new DSliderTrackHorizontal();
+	protected newTrackSelected( options?: OPTIONS ): DSliderTrack {
+		return new DSliderTrackHorizontal( options && options.track );
 	}
 
-	protected assertSize( options: OPTIONS ): void {
-		if( options ) {
-			if( options.height ) {
-				this.setWidth( options.height );
-			} else if( options.size ) {
-				switch( options.size ) {
-					case DSliderSize.LARGE:
-						this.setWidth( 450 );
-						break;
-					case DSliderSize.SMALL:
-						this.setWidth( 150 );
-						break;
-					default:
-						this.setWidth( this.theme.getWidth() );
-						break;
-				}
-			} else {
-				this.setWidth( this.theme.getWidth() );
-			}
-		} else {
-			this.setWidth( this.theme.getWidth() );
-		}
-
-		this.track.setWidth( this.width );
+	protected assertSize(): void {
+		this.track.setWidth( this.width);
+		this.trackSelected.setHeight( this.track.height);
 		this.trackSelected.setWidth( this.track.width );
```

Comment:
When the DSlider#width is dynamically changed, need to change the track width.

Reply:
Sorry, I don't fully understand. DSlider#width is adjusted just once when init.

Reply:
After the DSlider initialization, when the DSlider#width is changed somehow, this.track.width is needed to be adjusted again. E.g., if the width of the DSlider is `"100%"` and the window is resized, DSlider#width changes dynamically.

```
new DSlider({
     width: "100%"
});
```

Also we can change the DSlider#width programatically.

```
const slider = new DSlider();
slider.width *= 2;
```

I think we need to adjust the `this.track.width` when `DSlider#width` changes.

Reply:
If you like, you can use an expression something like `"100%-10"` as a `this.track.width`.
```
this.track.setWidth( "100%-10" );
```
In this case, you do not need to adjust the `this.track.width` when the `DSlider#width` changes. `DBase` will do for you.

Also, you can use a function as follows:
```
this.track.setWidth(( p: number ) => p - 10);
```

https://github.com/winter-cardinal/winter-cardinal-ui/blob/cbcf1e3a4435c26c4dc0e92c06ee758206df5b53/src/main/typescript/wcardinal/ui/d-base.ts#L1066-L1076

https://github.com/winter-cardinal/winter-cardinal-ui/blob/cbcf1e3a4435c26c4dc0e92c06ee758206df5b53/src/main/typescript/wcardinal/ui/d-coordinate.ts#L29

Reply:
If you really want to do something more complicated when the `DSlider#width` changes, please override the method `DBase#onParentResize`.
```
class DSlider {
    protected onParentResize( parentWidth: number, parentHeight: number, parentPadding: DPadding ): void {
        super.onParentResize( parentWidth, parentHeight, parentPadding );
        
        // DO SOMETHING HERE
       
    }
}
```
https://github.com/winter-cardinal/winter-cardinal-ui/blob/cbcf1e3a4435c26c4dc0e92c06ee758206df5b53/src/main/typescript/wcardinal/ui/d-base.ts#L1686-L1696

Reply:
> After the `DSlider` initialization, when the `DSlider#width` is changed somehow, `this.track.width` is needed to be adjusted again. E.g., if the width of the `DSlider` is `"100%"` and the window is resized, `DSlider#width` changes dynamically.

I don't know that DSlider can be scaled when window is resized.
Now I understood. 

So we can change to following:
protected assertSize(): void {
	this.track.setWidth( "100%" );
	this.trackSelected.setHeight( this.track.height);
	this.trackSelected.setWidth( "100%" );
}
Right?

Reply:
If we can assume that the height never changes, yes, something like that.

Reply:
Or we can also do something like following to adjust the size:
DSlider#onParentResize
```typescript
onParentResize( parentWidth: number, parentHeight: number, parentPadding: DPadding ): void {
        super.onParentResize( parentWidth, parentHeight, parentPadding );
	// Adjust children's size
	this.assertSize();
    }
```
I think it is better. Because DSliderVertical is a bit complicated than DSliderHorizontal. Right? 
```typescript
this.track.setHeight( this.height - this._value.height - VERTICAL_PIXEL_BALANCE );
this.trackSelected.setHeight( this.track.height - VERTICAL_PIXEL_BALANCE );
this.trackSelected.setWidth( this.track.width );
```

Reply:
I need to correct one thing I just said. Please override the `DBase#onResize` instead of the `DBase#onParentResize`. 

```
class DSlider {
    protected onResize( newWidth: number, newHeight: number, oldWidth: number, oldHeight: number ): void {
        super.onResize(  newWidth, newHeight, oldWidth, oldHeight );
        
        // DO SOMETHING HERE
       
    }
}
```

> I think it is better. Because DSliderVertical is a bit complicated than DSliderHorizontal.
> Right?

I think so too.